src/agent/onefuzz-task/src/tasks/coverage/dotnet.rs

```diff
@@ -0,0 +1,418 @@
+        info!("starting dotnet_coverage task");
+        self.config.coverage.init_pull().await?;
+
+        let dotnet_path = dotnet_path()?;
+        let dotnet_coverage_path = dotnet_coverage_path()?;
+
+        let heartbeat = self.config.common.init_heartbeat(None).await?;
+        let mut context = TaskContext::new(
+            &self.config,
+            heartbeat,
+            dotnet_path,
+            dotnet_coverage_path.clone(),
+        );
+
+        if !context.uses_input() {
+            bail!("input is not specified on the command line or arguments for the target");
+        }
+
+        context.heartbeat.alive();
+
+        let coverage_local_path = self.config.coverage.local_path.canonicalize()?;
+        let intermediate_files_path = intermediate_coverage_files_path(&coverage_local_path)?;
+        fs::create_dir_all(&intermediate_files_path).await?;
+        let timeout = self.config.timeout();
+        let coverage_dir = self.config.coverage.clone();
+        let dotnet_coverage_path = dotnet_coverage_path;
+
+        tokio::spawn(async move {
+            if let Err(e) = start_directory_monitor(
+                &intermediate_files_path,
+                &coverage_local_path,
+                timeout,
+                coverage_dir,
+                &dotnet_coverage_path,
+            )
+            .await
+            {
+                error!("Directory monitor failed: {}", e);
+            }
+        });
+
+        for dir in &self.config.readonly_inputs {
+            debug!("recording coverage for {}", dir.local_path.display());
+
+            dir.init_pull().await?;
+            let dir_count = context.record_corpus(&dir.local_path).await?;
+
+            info!(
+                "recorded coverage for {} inputs from {}",
+                dir_count,
+                dir.local_path.display()
+            );
+
+            context.heartbeat.alive();
+        }
+
+        context.heartbeat.alive();
+
+        if let Some(queue) = &self.config.input_queue {
+            info!("polling queue for new coverage inputs");
+
+            let callback = CallbackImpl::new(queue.clone(), context)?;
+            self.poller.run(callback).await?;
+        }
+
+        Ok(())
+    }
+}
+
+async fn start_directory_monitor(
+    intermediate_files_path: &PathBuf,
+    coverage_local_path: &Path,
+    timeout: Duration,
+    coverage_dir: SyncedDir,
+    dotnet_coverage_path: &Path,
+) -> Result<()> {
+    info!("Starting directory monitor");
+    let mut monitor = DirectoryMonitor::new(intermediate_files_path).await?;
+    debug!("Started directory monitor, waiting for files");
+    while (monitor.next_file().await?).is_some() {
+        debug!("Found intermediate coverage file");
+        save_and_sync_coverage(
+            coverage_local_path,
+            timeout,
+            coverage_dir.clone(),
+            dotnet_coverage_path,
+        )
+        .await?;
+        info!("Merged and synced coverage");
```

Comment:
"Merged" has an additional special meaning in the context of fuzzing (i.e. corpus merging), and though your usage is absolutely not wrong (and unambiguous here), I'd still suggest:
```suggestion
        info!("Updated and synced coverage");
```